File: testutils/src/main/java/com/google/firebase/iid/FirebaseInstanceId.java
package com.google.firebase.iid;

public class FirebaseInstanceId {
    private static String token;

    public static FirebaseInstanceId getInstance() {
        return new FirebaseInstanceId();
    }

    public static void setToken(String token) {
        FirebaseInstanceId.token = token;
    }

    public String getToken() {
        return token;
    }

    public String getToken(String authority, String scope) {
        return token;
    }
}

File: testutils/src/main/java/com/google/firebase/iid/FirebaseInstanceId.kt
package com.google.firebase.iid

object FirebaseInstanceIdToken {
    var token: String? = null
}

class FirebaseInstanceId {

    companion object {
        @JvmStatic
        fun getInstance() = FirebaseInstanceId()

        @JvmStatic
        fun setToken(token: String?) {
            FirebaseInstanceIdToken.token = token
        }
    }

    fun getToken() = FirebaseInstanceIdToken.token
    fun getToken(authority: String, scope: String) = FirebaseInstanceIdToken.token
}
